src/Components/ThemesMenu/ThemesMenu.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from "react";
+import Button from "../Button/Button";
+import "./ThemesMenu.css";
+
+export function ThemesMenu({
+  currentTheme,
+  onChange,
+  themes,
+  onAddTheme,
+  onEditTheme,
+  onDeleteTheme,
+}) {
+  const [newThemeName, setNewThemeName] = useState("");
+  const [isEditingTheme, setIsEditingTheme] = useState(false);
+  const [editThemeName, setEditThemeName] = useState(currentTheme.name);
+  const [isDelitingTheme, setIsDeletingTheme] = useState(false);
+
```

Comment:
just a suggestion you can also add single useState  here :
 ` const [themeState, setThemeState] = useState({

    isAddingTheme: false,

    isEditingTheme: false,

    isDeletingTheme: false,

    newThemeName: "",

    editThemeName: currentTheme.name,

  });`